src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml

```diff
@@ -29,7 +29,11 @@
     <type fullname="System.Runtime.CompilerServices.AsyncVoidMethodBuilder">
       <property name="ObjectIdForDebugger" />
     </type>
-    <type fullname="System.Runtime.CompilerServices.AsyncTaskMethodBuilder*">
+    <type fullname="System.Runtime.CompilerServices.AsyncTaskMethodBuilder">
```

Comment:
@stephentoub - I see that we also have:

* `AsyncValueTaskMethodBuilder`
* `AsyncValueTaskMethodBuilder<TResult>`

types, but they aren't getting their `ObjectIdForDebugger` preserved. Is this intended?

Reply:
We should probably put them on the same path as AsyncTaskMethodBuilder.  However, last I knew, the debugger didn't actually pay attention to these for AsyncValueTaskMethodBuilder, only to the ones for AsyncTaskMethodBuilder, so it hasn't really mattered.  With the increase in ValueTask usage, it'd be nice for .NET 5 to be able to support any improvements in the debugger here.

Reply:
Preserving these new types with https://github.com/dotnet/runtime/pull/39084.